warehouse/query-core/src/main/java/datawave/query/tables/ShardQueryLogic.java

```diff
@@ -1350,9 +1355,32 @@ protected void loadQueryModel(MetadataHelper helper, ShardQueryConfiguration con
     }
 
     protected Scheduler getScheduler(ShardQueryConfiguration config, ScannerFactory scannerFactory) {
+        // ony optimize if unique fields are not requested
+        if (isUseDocumentScheduler() && config.getUniqueFields().isEmpty() && !isCheckpointable() && !isFullTableScanEnabled()) {
```

Comment:
it looks like there may be other nodes besides unique fields (like functions and query markers?) that won't be supported by the SimpleQueryVisitor. Add defeats for those as well?